### PR TITLE
Add spec to update

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by JNKPercona** [comment]
PSMDB operator docker  - perconalab/percona-xtradb-cluster-operator:PR-216-3508225

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 1]
```diff
@@ -427,7 +427,7 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(arbiter bool, cr *a
 		if err != nil {
 			return nil, fmt.Errorf("PodDisruptionBudget for %s: %v", sfs.Name, err)
 		}
-
+		sfs.Spec = sfsSpec
 		sfs.Spec.Replicas = &size
 		sfs.Spec.Template.Spec.Containers = sfsSpec.Template.Spec.Containers
 		sfs.Spec.Template.Spec.Volumes = sfsSpec.Template.Spec.Volumes
```
Comment (dAdAbird) [inline pkg/controller/perconaservermongodb/psmdb_controller.go]: so there is no need in L431-433 anymore. as well as the duplication of the same code (`sfs.Spec = sfsSpec`) in the `if` above (L451). Just put it before `if-else`, so `if-else` is just decides to make create or update call.

Comment (dAdAbird) [reply pkg/controller/perconaservermongodb/psmdb_controller.go]: ok, `sfs.Spec.Replicas = &size` we need. but again, I suggest to move if outside of the `if-else`

**Comment by JNKPercona** [comment]
PSMDB operator docker  - perconalab/percona-xtradb-cluster-operator:PR-216-2105318

**Pull request merged**